src/subtitling/Group setups/github.md

```diff
@@ -0,0 +1,145 @@
+Here you can add users or teams (in an organization) to give access to your repository.
+You'll want to give them either "write" or "maintain" roles
+so they'll be able to properly make changes without many headaches.
+
+![Managing access](../images/gh_manage_access.png)
+
+
+[GitHub_home]: (https://github.com/)
+[GitLab_home]: (https://about.gitlab.com/)
+[BitBucket_home]: (https://bitbucket.org/)
+
+## Basic actions
+
+**If you are new to GitHub,
+we highly recommend you use the [GitHub desktop client](GitHub_desktop)**
+(this also works with other,
+similar platforms).
+This guide was written under the assumption that most users will be using this client.
+
+### Cloning
+
+After installing the desktop client,
+open it and log in.
+You can then clone the repository to your system by going to `File > Clone repository`,
+or by going to the repository on GitHub and going `Code > Open with GitHub Desktop`.
+
+![Opening a repo with the desktop client](../images/gh_clone.png)
+
+If the repository has not been set up yet,
+you can set it up in your desktop client through the link on the repository.
+This will automatically open up the cloning menu in your desktop client.
+
+![Setting up a repo with the desktop client](../images/gh_setup_repo.png)
+
+### Syncing file changes
+
+When editing files from a Git instance,
+Git will remember and verify what files have been changed.
+Before making any commits however,
+you'll want to first double-check that you have every file pulled to your local repository.
+Click "Fetch origin" at the top,
+and pull in any newly-pushed commits made by other users.
+
+Now you can verify your own changes and commit the files you want to upload/update in the repository.
+The desktop client will additionally give you a very useful display of what is different between your file
+and the latest version of the file you pulled in from the repo.
+To make things easier for your teammates,
+you'll want to make sure to give your commit a helpful title and description if necessary.
+Finally,
+press "Commit to **master**",
+and once you've finished making all your changes and committed them,
+press "Push origin".
+
+![Pulling in changes made to the repository](../images/gh_syncing.png)
+
+
+If you want to check the history to see what your teammates have changed,
+you can view the history by clicking the "History" tab.
+This will give you a useful oversight of all the commits,
+as well as their titles/descriptions and what changes have been made to the files.
+
+![Viewing the repository's history](../images/gh_history.png)
+
+[GitHub_desktop]: (https://desktop.github.com/)
+
+## Resolving merge conflicts
+
+There are going to be times when multiple users accidentally write to the same file at the same time.
+This is usually no issue,
+but you may from time to time run into a merge conflict.
+This means that changes made by multiple users are incompatible with one another,
+and will require manual fixing.
```

Comment:
Unless you set up a filter for the Aegisub Project Garbage section, you will have merge conflicts for *every concurrent change*. We should definitely include this filter somwhere here, but the easiest way I know of uses bash, which obviously wouldn't work on Windows.

Reply:
That depends largely on your chain of operations. For example, the way I have it set up:
1. The translator translates on a pad
2. The timer times that to a script
3. The timer pushes the script to the repo
4. Editor/TLC do their job (in the order they arranged between themselves)
5. Typesetting is done in a separate file
6. QC does work

This is something I was intending to cover more in-depth in the SubKt guide, since that's where you can set up auto-merging and the like, but I may add a small bit about it here as well.

Reply:
Sure, if you work sequentially, you don't run into problems with concurrent changes. Although it's very easy to forget to pull before making changes.

I generally find Aegisub Project Garbage changes in my git history to be annoying since they are 100% redundant and have no place in a shared git repo, but it's not required, of course.